### PR TITLE
Code for SSL implementation for API/GUI

### PR DESCRIPTION
**Pull request by aman566**
This is the code for implementing SSL module for API/GUI for nettacker. Solved issue #133.
Command which will run the server up on https:
`python nettacker.py -i google.com --start-api --api-host 127.0.0.1 --api-port 9999`

sslimplemented

#### Checklist
- [X] I have followed the [Contributor Guidelines](https://github.com/zdresearch/OWASP-Nettacker/wiki/Developers#contribution-guidelines).
- [X] I have added the relevant documentation.
- [X] My branch is up-to-date with the Upstream master branch.

#### Your development environment
- OS: `LInux`
- OS Version: `Ubuntu`
- Python Version: `3.x`

### CODE REVIEW

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- core/args_loader.py  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770&bid=17004097)

**Comment by aman566** [comment]
Hello @Ali-Razmjoo I have merged all the code for SSL into this single pull request. please help me in fixing this Codacy issue. It worked will in my local machine.

**Comment by aman566** [comment]
Hello @Ali-Razmjoo , my work for the issue #133 is completed. Please review it. the api/gui is now running on https. Will be waiting for a response :)

**Comment by Ali-Razmjoo** [comment]
@securestep9 can you review please?

File: lib/vuln/http_cookie/engine.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+import socket
+import socks
+import time
+import json
+import threading
+import string
+import random
+import sys
+import struct
+import re
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused import re](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  4
- lib/vuln/http_cookie/engine.py  23
         

Clones added
============
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  5
- lib/vuln/http_cookie/engine.py  52
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770&bid=17004097)

File: lib/vuln/http_cookie/engine.py [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+import socket
+import socks
+import time
+import json
+import threading
+import string
+import random
+import sys
+import struct
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused import struct](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            if "samesite" in value:
+                return False
+            if "secure" in value:
+                return False
+            if "httponly" in value:
+                return False
+            else:
+                return True
+
+    except Exception as e:
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused variable 'e'](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+import time
+import json
+import threading
+import string
+import random
+import sys
+import struct
+import re
+import os
+from OpenSSL import crypto
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused crypto imported from OpenSSL](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+class HTTP_CookieVulnScan(Transform):
+    """TODO: Your transform description."""
+
+    # The transform input entity type.
+    input_type = NettackerScan
+
+    def do_transform(self, request, response, config):
+        # TODO: write your code here.
+        scan_request = request.entity
+        scan_id = "".join(random.choice("0123456789abcdef") for x in range(32))
```
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+
+@EnableDebugWindow
+class HTTP_CookieVulnScan(Transform):
+    """TODO: Your transform description."""
+
+    # The transform input entity type.
+    input_type = NettackerScan
+
+    def do_transform(self, request, response, config):
```
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+                        extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["http_cookie_vuln_ports"]
+        if target_type(target) == 'HTTP':
+            target = target_to_host(target)
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+import json
+import threading
+import string
+import random
+import sys
+import struct
+import re
+import os
+from OpenSSL import crypto
+import ssl
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused import ssl](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- lib/vuln/http_cookie/engine.py  23
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  4
         

Clones added
============
- lib/vuln/http_cookie/engine.py  48
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770&bid=17004097)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 5 above)
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 6 above)
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+                        extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["http_cookie_vuln_ports"]
+        if target_type(target) == 'HTTP':
+            target = target_to_host(target)
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 10]
```diff
@@ -0,0 +1,170 @@
...
+            if "samesite" in value:
+                return False
+            if "secure" in value:
+                return False
+            if "httponly" in value:
+                return False
+            else:
+                return True
+
+    except Exception as e:
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused variable 'e'](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- lib/vuln/http_cookie/engine.py  23
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  4
         

Clones added
============
- lib/vuln/http_cookie/engine.py  48
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770&bid=17004097)

File: lib/vuln/http_cookie/engine.py [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+                        extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["http_cookie_vuln_ports"]
+        if target_type(target) == 'HTTP':
+            target = target_to_host(target)
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 5 above)
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 6 above)
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+            if "samesite" in value:
+                return False
+            if "secure" in value:
+                return False
+            if "httponly" in value:
+                return False
+            else:
+                return True
+
+    except Exception as e:
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused variable 'e'](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

**Comment by aman566** [comment]
Hello @Ali-Razmjoo I have added a new module for finding set-cookie header based vulnerabilities.
I don't know why codacy is breaking. Please help me regarding this.

set-cookie

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  4
- lib/vuln/http_cookie/engine.py  23
         

Clones added
============
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  5
- lib/vuln/http_cookie/engine.py  48
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770&bid=17004097)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 6 above)
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Method could be a function](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+                        extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["http_cookie_vuln_ports"]
+        if target_type(target) == 'HTTP':
+            target = target_to_host(target)
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 5 above)
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+            if target_type(target) != "HTTP" and port == 80:
+                target = 'http://' + target
+            req = requests.get(target)
+            value = req.headers['Set-Cookie'].lower()
+            if "samesite" in value:
+                return False
+            else:
+                return True
+
+    except Exception as e:
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused variable 'e'](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Complexity increasing per file
==============================
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  4
- lib/vuln/http_cookie/engine.py  23
         

Clones added
============
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  5
- lib/vuln/http_cookie/engine.py  48
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770&bid=17004097)

File: lib/vuln/http_cookie/engine.py (hunk 14 above)
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Unused variable 'e'](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py [hunk 15]
```diff
@@ -0,0 +1,49 @@
...
+class HTTP_CookieVulnScan(Transform):
+    """TODO: Your transform description."""
+
+    # The transform input entity type.
+    input_type = NettackerScan
+    @staticmethod
+    def do_transform(self, request, response, config):
+        # TODO: write your code here.
+        scan_request = request.entity
+        scan_id = "".join(random.choice("0123456789abcdef") for x in range(32))
```
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py (hunk 13 above)
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- lib/vuln/http_cookie/engine.py  23
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  4
         

Clones added
============
- lib/vuln/http_cookie/engine.py  48
- lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770&bid=17004097)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 15 above)
Comment (Ali-Razmjoo) [inline lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/vuln/http_cookie/engine.py [hunk 16]
```diff
@@ -0,0 +1,166 @@
...
+                        extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["http_cookie_vuln_ports"]
+        if target_type(target) == 'HTTP':
+            target = target_to_host(target)
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
```
Comment (Ali-Razmjoo) [inline lib/vuln/http_cookie/engine.py]: Codacy Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)

File: lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py (hunk 15 above)
Comment (aman566) [reply lib/transactions/maltego/nettacker_transforms/src/nettacker_transforms/transforms/http_cookie_vuln_scan.py]: I am searching for this issue and will try to fix it.

**Comment by aman566** [comment]
Closing this PR for a more cleaner commit history. Added two seperate PR's for 2 separate modules

**Pull request closed without merging**